{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/mol/Bioinfo_Routine/Variant_Vault/_pipeline\n"
     ]
    }
   ],
   "source": [
    "%cd /mnt/mol/Bioinfo_Routine/Variant_Vault/_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "import hashlib\n",
    "\n",
    "def generate_unique_id(data):\n",
    "    # Create a hash object\n",
    "    hash_object = hashlib.sha256(data.encode())\n",
    "    \n",
    "    # Convert the hash to a hexadecimal string\n",
    "    hex_dig = hash_object.hexdigest()\n",
    "    \n",
    "    # Convert hexadecimal to integer (you can also truncate this to fit your needs)\n",
    "    unique_id = str(int(hex_dig, 16))\n",
    "    \n",
    "    return unique_id\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Rekap_TSHC_MAF_20230414.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220816.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230116.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230102.xlsx',\n",
       " 'Rekap_TSHC_MAF_20210804.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230313.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220224.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230515.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231120.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230605.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230306.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220819.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230717.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230502.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230424.xlsx',\n",
       " 'Rekap_TSHC_MAF_20240119.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231016.xlsx',\n",
       " 'Rekap_TSHC_MAF_20240205.xlsx',\n",
       " 'Rekap_TSHC_MAF_20221121.xlsx',\n",
       " 'Rekap_TSHC_MAF_20240304.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220919.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220801.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230807.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230626.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230703.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220608.xlsx',\n",
       " 'Rekap_TSHC_MAF_20210610.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230320.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231218.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220622.xlsx',\n",
       " 'Rekap_TSHC_MAF_20221019.xlsx',\n",
       " 'Rekap_TSHC_MAF_20221003.xlsx',\n",
       " 'Rekap_TSHC_MAF_20210624.xlsx',\n",
       " 'Rekap_TSHC_MAF_20210714.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230712.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230516.xlsx',\n",
       " 'Rekap_TSHC_MAF_20221219.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230331.xlsx',\n",
       " 'Rekap_TSHC_MAF_20221205.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230918.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231204.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231122.xlsx',\n",
       " 'Rekap_TSHC_MAF_20221107.xlsx',\n",
       " 'Rekap_TSHC_MAF_20240108.xlsx',\n",
       " 'Rekap_TSHC_MAF_20240219.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220519.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230823.xlsx',\n",
       " 'Rekap_TSHC_MAF_20210723.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220720.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231002.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230706.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231206.xlsx',\n",
       " 'Rekap_TSHC_MAF_20231106.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230206.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230220.xlsx',\n",
       " 'Rekap_TSHC_MAF_20230904.xlsx',\n",
       " 'Rekap_TSHC_MAF_20210707.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220421.xlsx',\n",
       " 'Rekap_TSHC_MAF_20220808.xlsx',\n",
       " 'Rekap_TSHC_MAF_20240117.xlsx']"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "from PBTK import *\n",
    "import pandas as pd\n",
    "from sqlalchemy import create_engine, Column, String, MetaData, Table\n",
    "import sqlite3\n",
    "\n",
    "vava_db_dir = '/home/sysadmin/vava_db/'\n",
    "\n",
    "\n",
    "current_directory = os.getcwd()\n",
    "# Get the parent directory\n",
    "parent_directory = os.path.dirname(current_directory)\n",
    "input_temp_dir = os.path.join(os.getcwd(),'input_temp')\n",
    "excel_path = os.path.join(current_directory, 'bahan_db','Rekap_TSHC_MAF_20240304.xlsx')\n",
    "# Read the Excel file\n",
    "excel_data = pd.read_excel(os.path.join(current_directory,excel_path))\n",
    "bahan_db = os.listdir(os.path.join(current_directory, 'bahan_db_brca',))\n",
    "bahan_db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "60"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(bahan_db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rekap_TSHC_MAF_20230414.xlsx\n",
      "Rekap_TSHC_MAF_20220816.xlsx\n",
      "Rekap_TSHC_MAF_20230116.xlsx\n",
      "Rekap_TSHC_MAF_20230102.xlsx\n",
      "Rekap_TSHC_MAF_20210804.xlsx\n",
      "Rekap_TSHC_MAF_20230313.xlsx\n",
      "Rekap_TSHC_MAF_20220224.xlsx\n",
      "Rekap_TSHC_MAF_20230515.xlsx\n",
      "Rekap_TSHC_MAF_20231120.xlsx\n",
      "Rekap_TSHC_MAF_20230605.xlsx\n",
      "Rekap_TSHC_MAF_20230306.xlsx\n",
      "Rekap_TSHC_MAF_20220819.xlsx\n",
      "Rekap_TSHC_MAF_20230717.xlsx\n",
      "Rekap_TSHC_MAF_20230502.xlsx\n",
      "Rekap_TSHC_MAF_20230424.xlsx\n",
      "Rekap_TSHC_MAF_20240119.xlsx\n",
      "Rekap_TSHC_MAF_20231016.xlsx\n",
      "Rekap_TSHC_MAF_20240205.xlsx\n",
      "Rekap_TSHC_MAF_20221121.xlsx\n",
      "Rekap_TSHC_MAF_20240304.xlsx\n",
      "Rekap_TSHC_MAF_20220919.xlsx\n",
      "Rekap_TSHC_MAF_20220801.xlsx\n",
      "Rekap_TSHC_MAF_20230807.xlsx\n",
      "Rekap_TSHC_MAF_20230626.xlsx\n",
      "Rekap_TSHC_MAF_20230703.xlsx\n",
      "Rekap_TSHC_MAF_20220608.xlsx\n",
      "Rekap_TSHC_MAF_20210610.xlsx\n",
      "Rekap_TSHC_MAF_20230320.xlsx\n",
      "Rekap_TSHC_MAF_20231218.xlsx\n",
      "Rekap_TSHC_MAF_20220622.xlsx\n",
      "Rekap_TSHC_MAF_20221019.xlsx\n",
      "Rekap_TSHC_MAF_20221003.xlsx\n",
      "Rekap_TSHC_MAF_20210624.xlsx\n",
      "Rekap_TSHC_MAF_20210714.xlsx\n",
      "Rekap_TSHC_MAF_20230712.xlsx\n",
      "Rekap_TSHC_MAF_20230516.xlsx\n",
      "Rekap_TSHC_MAF_20221219.xlsx\n",
      "Rekap_TSHC_MAF_20230331.xlsx\n",
      "Rekap_TSHC_MAF_20221205.xlsx\n",
      "Rekap_TSHC_MAF_20230918.xlsx\n",
      "Rekap_TSHC_MAF_20231204.xlsx\n",
      "Rekap_TSHC_MAF_20231122.xlsx\n",
      "Rekap_TSHC_MAF_20221107.xlsx\n",
      "Rekap_TSHC_MAF_20240108.xlsx\n",
      "Rekap_TSHC_MAF_20240219.xlsx\n",
      "Rekap_TSHC_MAF_20220519.xlsx\n",
      "Rekap_TSHC_MAF_20230823.xlsx\n",
      "Rekap_TSHC_MAF_20210723.xlsx\n",
      "Rekap_TSHC_MAF_20220720.xlsx\n",
      "Rekap_TSHC_MAF_20231002.xlsx\n",
      "Rekap_TSHC_MAF_20230706.xlsx\n",
      "Rekap_TSHC_MAF_20231206.xlsx\n",
      "Rekap_TSHC_MAF_20231106.xlsx\n",
      "Rekap_TSHC_MAF_20230206.xlsx\n",
      "Rekap_TSHC_MAF_20230220.xlsx\n",
      "Rekap_TSHC_MAF_20230904.xlsx\n",
      "Rekap_TSHC_MAF_20210707.xlsx\n",
      "Rekap_TSHC_MAF_20220421.xlsx\n",
      "Rekap_TSHC_MAF_20220808.xlsx\n",
      "Rekap_TSHC_MAF_20240117.xlsx\n"
     ]
    }
   ],
   "source": [
    "bahan_db_df = pd.DataFrame()\n",
    "bahan_db_df\n",
    "\n",
    "# Loop through the files, read each into a DataFrame, and append it to the merged_df\n",
    "for file in bahan_db:\n",
    "    print(file)\n",
    "    file_path = os.path.join(current_directory, 'bahan_db', file)\n",
    "    df_temp = pd.read_excel(file_path, engine='openpyxl')\n",
    "    bahan_db_df = pd.concat([bahan_db_df, df_temp], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Nucleotide [AAchange]</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>Clinical Relevance</th>\n",
       "      <th>P/LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "      <th>Disease_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7</td>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12</td>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>16</td>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>23</td>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>chr1</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>1780</td>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>chr22</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>1783</td>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr22</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not detected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>1784</td>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr22</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not detected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>1787</td>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr22</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not detected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>1788</td>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr22</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not detected</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Unnamed: 0     Rundate     SampleID   Gene  \\\n",
       "0                0  14/04/2023  R2303061875   SDHB   \n",
       "1                7  14/04/2023  R2303061875   SDHB   \n",
       "2               12  14/04/2023  R2303061875   SDHB   \n",
       "3               16  14/04/2023  R2303061875   SDHB   \n",
       "4               23  14/04/2023  R2303061875  MUTYH   \n",
       "...            ...         ...          ...    ...   \n",
       "162647        1780  17/01/2024  R2401062088  CHEK2   \n",
       "162648        1783  17/01/2024  R2401062088  CHEK2   \n",
       "162649        1784  17/01/2024  R2401062088  CHEK2   \n",
       "162650        1787  17/01/2024  R2401062088  CHEK2   \n",
       "162651        1788  17/01/2024  R2401062088  CHEK2   \n",
       "\n",
       "                                    Nucleotide [AAchange]    dbSNP ID  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--         NaN   \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--         NaN   \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--         NaN   \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A   rs2746462   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--   rs3219493   \n",
       "...                                                   ...         ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--  rs17886236   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--         NaN   \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--         NaN   \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--         NaN   \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...         NaN   \n",
       "\n",
       "       Chromosome     Start       End Ref Alt Variant Type  \\\n",
       "0            chr1  17032891  17032891   T   C          SNP   \n",
       "1            chr1  17044657  17044657   T   C          SNP   \n",
       "2            chr1  17044999  17044999   C   T          SNP   \n",
       "3            chr1  17054002  17054002   G   T          SNP   \n",
       "4            chr1  45330597  45330597   G   C          SNP   \n",
       "...           ...       ...       ...  ..  ..          ...   \n",
       "162647      chr22  28694987  28694987   C   G          SNP   \n",
       "162648      chr22  28695049  28695049   G   C          SNP   \n",
       "162649      chr22  28703766  28703766   T   C          SNP   \n",
       "162650      chr22  28734312  28734312   C   T          SNP   \n",
       "162651      chr22  28734825  28734826   -   A          INS   \n",
       "\n",
       "       Variant_Classification     VAF Genotype  alt_count  ref_count   DP  \\\n",
       "0                      Intron   49.74      Het         96         97  193   \n",
       "1                      Intron   45.91      Het         73         86  159   \n",
       "2                      Intron   44.70      Het         97        120  217   \n",
       "3                      Silent   99.66   HomAlt        876          3  879   \n",
       "4                      Intron  100.00   HomAlt        545          0  545   \n",
       "...                       ...     ...      ...        ...        ...  ...   \n",
       "162647                 Intron   68.18      Het         30         14   44   \n",
       "162648                 Intron   58.62      Het         51         36   87   \n",
       "162649                 Intron   46.67      Het          7          8   15   \n",
       "162650                 Intron   31.71      Het         13         28   41   \n",
       "162651                 Intron   50.00      Het         13         13   26   \n",
       "\n",
       "       Clinical Relevance   P/LP_Result Mode_of_Inheritance  \\\n",
       "0                     NaN  not detected                  AD   \n",
       "1                     NaN  not detected                  AD   \n",
       "2                     NaN  not detected                  AD   \n",
       "3                  Benign  not detected                  AD   \n",
       "4                  Benign  not detected                  AR   \n",
       "...                   ...           ...                 ...   \n",
       "162647      Likely_benign  not detected                 NaN   \n",
       "162648                NaN  not detected                 NaN   \n",
       "162649                NaN  not detected                 NaN   \n",
       "162650                NaN  not detected                 NaN   \n",
       "162651                NaN  not detected                 NaN   \n",
       "\n",
       "                         Disease_Name  \n",
       "0       Paragangliomas 4 (MIM 115310)  \n",
       "1       Paragangliomas 4 (MIM 115310)  \n",
       "2       Paragangliomas 4 (MIM 115310)  \n",
       "3       Paragangliomas 4 (MIM 115310)  \n",
       "4          Pilomatrixoma (MIM 132600)  \n",
       "...                               ...  \n",
       "162647                            NaN  \n",
       "162648                            NaN  \n",
       "162649                            NaN  \n",
       "162650                            NaN  \n",
       "162651                            NaN  \n",
       "\n",
       "[162652 rows x 22 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Nucleotide [AAchange]</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>Clinical Relevance</th>\n",
       "      <th>P/LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "      <th>Disease_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr1</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr1</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>chr1</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>chr1</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>chr22</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Rundate     SampleID   Gene  \\\n",
       "0       14/04/2023  R2303061875   SDHB   \n",
       "1       14/04/2023  R2303061875   SDHB   \n",
       "2       14/04/2023  R2303061875   SDHB   \n",
       "3       14/04/2023  R2303061875   SDHB   \n",
       "4       14/04/2023  R2303061875  MUTYH   \n",
       "...            ...          ...    ...   \n",
       "162647  17/01/2024  R2401062088  CHEK2   \n",
       "162648  17/01/2024  R2401062088  CHEK2   \n",
       "162649  17/01/2024  R2401062088  CHEK2   \n",
       "162650  17/01/2024  R2401062088  CHEK2   \n",
       "162651  17/01/2024  R2401062088  CHEK2   \n",
       "\n",
       "                                    Nucleotide [AAchange]    dbSNP ID  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--               \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--               \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--               \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A   rs2746462   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--   rs3219493   \n",
       "...                                                   ...         ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--  rs17886236   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--               \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--               \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--               \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...               \n",
       "\n",
       "       Chromosome     Start       End Ref Alt Variant Type  \\\n",
       "0            chr1  17032891  17032891   T   C          SNP   \n",
       "1            chr1  17044657  17044657   T   C          SNP   \n",
       "2            chr1  17044999  17044999   C   T          SNP   \n",
       "3            chr1  17054002  17054002   G   T          SNP   \n",
       "4            chr1  45330597  45330597   G   C          SNP   \n",
       "...           ...       ...       ...  ..  ..          ...   \n",
       "162647      chr22  28694987  28694987   C   G          SNP   \n",
       "162648      chr22  28695049  28695049   G   C          SNP   \n",
       "162649      chr22  28703766  28703766   T   C          SNP   \n",
       "162650      chr22  28734312  28734312   C   T          SNP   \n",
       "162651      chr22  28734825  28734826   -   A          INS   \n",
       "\n",
       "       Variant_Classification     VAF Genotype  alt_count  ref_count   DP  \\\n",
       "0                      Intron   49.74      Het         96         97  193   \n",
       "1                      Intron   45.91      Het         73         86  159   \n",
       "2                      Intron   44.70      Het         97        120  217   \n",
       "3                      Silent   99.66   HomAlt        876          3  879   \n",
       "4                      Intron  100.00   HomAlt        545          0  545   \n",
       "...                       ...     ...      ...        ...        ...  ...   \n",
       "162647                 Intron   68.18      Het         30         14   44   \n",
       "162648                 Intron   58.62      Het         51         36   87   \n",
       "162649                 Intron   46.67      Het          7          8   15   \n",
       "162650                 Intron   31.71      Het         13         28   41   \n",
       "162651                 Intron   50.00      Het         13         13   26   \n",
       "\n",
       "       Clinical Relevance   P/LP_Result Mode_of_Inheritance  \\\n",
       "0                          not detected                  AD   \n",
       "1                          not detected                  AD   \n",
       "2                          not detected                  AD   \n",
       "3                  Benign  not detected                  AD   \n",
       "4                  Benign  not detected                  AR   \n",
       "...                   ...           ...                 ...   \n",
       "162647      Likely_benign  not detected                       \n",
       "162648                     not detected                       \n",
       "162649                     not detected                       \n",
       "162650                     not detected                       \n",
       "162651                     not detected                       \n",
       "\n",
       "                         Disease_Name  \n",
       "0       Paragangliomas 4 (MIM 115310)  \n",
       "1       Paragangliomas 4 (MIM 115310)  \n",
       "2       Paragangliomas 4 (MIM 115310)  \n",
       "3       Paragangliomas 4 (MIM 115310)  \n",
       "4          Pilomatrixoma (MIM 132600)  \n",
       "...                               ...  \n",
       "162647                                 \n",
       "162648                                 \n",
       "162649                                 \n",
       "162650                                 \n",
       "162651                                 \n",
       "\n",
       "[162652 rows x 21 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bahan_db_df = bahan_db_df.drop(columns='Unnamed: 0')\n",
    "bahan_db_df.fillna(' ', inplace=True)\n",
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Nucleotide [AAchange]</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>Clinical Relevance</th>\n",
       "      <th>P/LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "      <th>Disease_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr1</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr1</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>chr1</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>chr1</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>chr1</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>Benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>chr22</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td></td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Rundate     SampleID   Gene  \\\n",
       "0       14/04/2023  R2303061875   SDHB   \n",
       "1       14/04/2023  R2303061875   SDHB   \n",
       "2       14/04/2023  R2303061875   SDHB   \n",
       "3       14/04/2023  R2303061875   SDHB   \n",
       "4       14/04/2023  R2303061875  MUTYH   \n",
       "...            ...          ...    ...   \n",
       "162647  17/01/2024  R2401062088  CHEK2   \n",
       "162648  17/01/2024  R2401062088  CHEK2   \n",
       "162649  17/01/2024  R2401062088  CHEK2   \n",
       "162650  17/01/2024  R2401062088  CHEK2   \n",
       "162651  17/01/2024  R2401062088  CHEK2   \n",
       "\n",
       "                                    Nucleotide [AAchange]    dbSNP ID  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--               \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--               \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--               \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A   rs2746462   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--   rs3219493   \n",
       "...                                                   ...         ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--  rs17886236   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--               \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--               \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--               \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...               \n",
       "\n",
       "       Chromosome     Start       End Ref Alt Variant Type  \\\n",
       "0            chr1  17032891  17032891   T   C          SNP   \n",
       "1            chr1  17044657  17044657   T   C          SNP   \n",
       "2            chr1  17044999  17044999   C   T          SNP   \n",
       "3            chr1  17054002  17054002   G   T          SNP   \n",
       "4            chr1  45330597  45330597   G   C          SNP   \n",
       "...           ...       ...       ...  ..  ..          ...   \n",
       "162647      chr22  28694987  28694987   C   G          SNP   \n",
       "162648      chr22  28695049  28695049   G   C          SNP   \n",
       "162649      chr22  28703766  28703766   T   C          SNP   \n",
       "162650      chr22  28734312  28734312   C   T          SNP   \n",
       "162651      chr22  28734825  28734826   -   A          INS   \n",
       "\n",
       "       Variant_Classification     VAF Genotype  alt_count  ref_count   DP  \\\n",
       "0                      Intron   49.74      Het         96         97  193   \n",
       "1                      Intron   45.91      Het         73         86  159   \n",
       "2                      Intron   44.70      Het         97        120  217   \n",
       "3                      Silent   99.66   HomAlt        876          3  879   \n",
       "4                      Intron  100.00   HomAlt        545          0  545   \n",
       "...                       ...     ...      ...        ...        ...  ...   \n",
       "162647                 Intron   68.18      Het         30         14   44   \n",
       "162648                 Intron   58.62      Het         51         36   87   \n",
       "162649                 Intron   46.67      Het          7          8   15   \n",
       "162650                 Intron   31.71      Het         13         28   41   \n",
       "162651                 Intron   50.00      Het         13         13   26   \n",
       "\n",
       "       Clinical Relevance   P/LP_Result Mode_of_Inheritance  \\\n",
       "0                          not detected                  AD   \n",
       "1                          not detected                  AD   \n",
       "2                          not detected                  AD   \n",
       "3                  Benign  not detected                  AD   \n",
       "4                  Benign  not detected                  AR   \n",
       "...                   ...           ...                 ...   \n",
       "162647      Likely_benign  not detected                       \n",
       "162648                     not detected                       \n",
       "162649                     not detected                       \n",
       "162650                     not detected                       \n",
       "162651                     not detected                       \n",
       "\n",
       "                         Disease_Name  \n",
       "0       Paragangliomas 4 (MIM 115310)  \n",
       "1       Paragangliomas 4 (MIM 115310)  \n",
       "2       Paragangliomas 4 (MIM 115310)  \n",
       "3       Paragangliomas 4 (MIM 115310)  \n",
       "4          Pilomatrixoma (MIM 132600)  \n",
       "...                               ...  \n",
       "162647                                 \n",
       "162648                                 \n",
       "162649                                 \n",
       "162650                                 \n",
       "162651                                 \n",
       "\n",
       "[162652 rows x 21 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Nucleotide [AAchange]</th>\n",
       "      <th>Clinical Relevance</th>\n",
       "      <th>Disease Name</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>P/LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14/04/2023</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>17/01/2024</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Rundate     SampleID  \\\n",
       "0       14/04/2023  R2303061875   \n",
       "1       14/04/2023  R2303061875   \n",
       "2       14/04/2023  R2303061875   \n",
       "3       14/04/2023  R2303061875   \n",
       "4       14/04/2023  R2303061875   \n",
       "...            ...          ...   \n",
       "162647  17/01/2024  R2401062088   \n",
       "162648  17/01/2024  R2401062088   \n",
       "162649  17/01/2024  R2401062088   \n",
       "162650  17/01/2024  R2401062088   \n",
       "162651  17/01/2024  R2401062088   \n",
       "\n",
       "                                    Nucleotide [AAchange] Clinical Relevance  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--                      \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--                      \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--                      \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A             Benign   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--             Benign   \n",
       "...                                                   ...                ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--      Likely_benign   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--                      \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--                      \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--                      \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...                      \n",
       "\n",
       "                         Disease Name Chromosome    dbSNP ID   Gene     Start  \\\n",
       "0       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17032891   \n",
       "1       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044657   \n",
       "2       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044999   \n",
       "3       Paragangliomas 4 (MIM 115310)       chr1   rs2746462   SDHB  17054002   \n",
       "4          Pilomatrixoma (MIM 132600)       chr1   rs3219493  MUTYH  45330597   \n",
       "...                               ...        ...         ...    ...       ...   \n",
       "162647                                     chr22  rs17886236  CHEK2  28694987   \n",
       "162648                                     chr22              CHEK2  28695049   \n",
       "162649                                     chr22              CHEK2  28703766   \n",
       "162650                                     chr22              CHEK2  28734312   \n",
       "162651                                     chr22              CHEK2  28734825   \n",
       "\n",
       "             End Ref Alt Variant Type Variant_Classification     VAF Genotype  \\\n",
       "0       17032891   T   C          SNP                 Intron   49.74      Het   \n",
       "1       17044657   T   C          SNP                 Intron   45.91      Het   \n",
       "2       17044999   C   T          SNP                 Intron   44.70      Het   \n",
       "3       17054002   G   T          SNP                 Silent   99.66   HomAlt   \n",
       "4       45330597   G   C          SNP                 Intron  100.00   HomAlt   \n",
       "...          ...  ..  ..          ...                    ...     ...      ...   \n",
       "162647  28694987   C   G          SNP                 Intron   68.18      Het   \n",
       "162648  28695049   G   C          SNP                 Intron   58.62      Het   \n",
       "162649  28703766   T   C          SNP                 Intron   46.67      Het   \n",
       "162650  28734312   C   T          SNP                 Intron   31.71      Het   \n",
       "162651  28734826   -   A          INS                 Intron   50.00      Het   \n",
       "\n",
       "        alt_count  ref_count   DP   P/LP_Result Mode_of_Inheritance  \n",
       "0              96         97  193  not detected                  AD  \n",
       "1              73         86  159  not detected                  AD  \n",
       "2              97        120  217  not detected                  AD  \n",
       "3             876          3  879  not detected                  AD  \n",
       "4             545          0  545  not detected                  AR  \n",
       "...           ...        ...  ...           ...                 ...  \n",
       "162647         30         14   44  not detected                      \n",
       "162648         51         36   87  not detected                      \n",
       "162649          7          8   15  not detected                      \n",
       "162650         13         28   41  not detected                      \n",
       "162651         13         13   26  not detected                      \n",
       "\n",
       "[162652 rows x 21 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bahan_db_df.rename(columns={'Disease_Name': 'Disease Name'}, inplace=True)\n",
    "all_columns = bahan_db_df.columns\n",
    "desired_columns = ['Rundate','SampleID','Nucleotide [AAchange]', 'Clinical Relevance', 'Disease Name', 'Chromosome', 'dbSNP ID']\n",
    "new_column_order = [col for col in desired_columns if col in all_columns] + [col for col in all_columns if col not in desired_columns]\n",
    "new_column_order\n",
    "bahan_db_df = bahan_db_df[new_column_order]\n",
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Nucleotide [AAchange]</th>\n",
       "      <th>Clinical Relevance</th>\n",
       "      <th>Disease Name</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>P/LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Rundate     SampleID  \\\n",
       "0       2023/04/14  R2303061875   \n",
       "1       2023/04/14  R2303061875   \n",
       "2       2023/04/14  R2303061875   \n",
       "3       2023/04/14  R2303061875   \n",
       "4       2023/04/14  R2303061875   \n",
       "...            ...          ...   \n",
       "162647  2024/01/17  R2401062088   \n",
       "162648  2024/01/17  R2401062088   \n",
       "162649  2024/01/17  R2401062088   \n",
       "162650  2024/01/17  R2401062088   \n",
       "162651  2024/01/17  R2401062088   \n",
       "\n",
       "                                    Nucleotide [AAchange] Clinical Relevance  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--                      \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--                      \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--                      \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A             Benign   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--             Benign   \n",
       "...                                                   ...                ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--      Likely_benign   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--                      \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--                      \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--                      \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...                      \n",
       "\n",
       "                         Disease Name Chromosome    dbSNP ID   Gene     Start  \\\n",
       "0       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17032891   \n",
       "1       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044657   \n",
       "2       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044999   \n",
       "3       Paragangliomas 4 (MIM 115310)       chr1   rs2746462   SDHB  17054002   \n",
       "4          Pilomatrixoma (MIM 132600)       chr1   rs3219493  MUTYH  45330597   \n",
       "...                               ...        ...         ...    ...       ...   \n",
       "162647                                     chr22  rs17886236  CHEK2  28694987   \n",
       "162648                                     chr22              CHEK2  28695049   \n",
       "162649                                     chr22              CHEK2  28703766   \n",
       "162650                                     chr22              CHEK2  28734312   \n",
       "162651                                     chr22              CHEK2  28734825   \n",
       "\n",
       "             End Ref Alt Variant Type Variant_Classification     VAF Genotype  \\\n",
       "0       17032891   T   C          SNP                 Intron   49.74      Het   \n",
       "1       17044657   T   C          SNP                 Intron   45.91      Het   \n",
       "2       17044999   C   T          SNP                 Intron   44.70      Het   \n",
       "3       17054002   G   T          SNP                 Silent   99.66   HomAlt   \n",
       "4       45330597   G   C          SNP                 Intron  100.00   HomAlt   \n",
       "...          ...  ..  ..          ...                    ...     ...      ...   \n",
       "162647  28694987   C   G          SNP                 Intron   68.18      Het   \n",
       "162648  28695049   G   C          SNP                 Intron   58.62      Het   \n",
       "162649  28703766   T   C          SNP                 Intron   46.67      Het   \n",
       "162650  28734312   C   T          SNP                 Intron   31.71      Het   \n",
       "162651  28734826   -   A          INS                 Intron   50.00      Het   \n",
       "\n",
       "        alt_count  ref_count   DP   P/LP_Result Mode_of_Inheritance  \n",
       "0              96         97  193  not detected                  AD  \n",
       "1              73         86  159  not detected                  AD  \n",
       "2              97        120  217  not detected                  AD  \n",
       "3             876          3  879  not detected                  AD  \n",
       "4             545          0  545  not detected                  AR  \n",
       "...           ...        ...  ...           ...                 ...  \n",
       "162647         30         14   44  not detected                      \n",
       "162648         51         36   87  not detected                      \n",
       "162649          7          8   15  not detected                      \n",
       "162650         13         28   41  not detected                      \n",
       "162651         13         13   26  not detected                      \n",
       "\n",
       "[162652 rows x 21 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Convert the 'Date' column to datetime format using the 'dayfirst' parameter\n",
    "bahan_db_df['Rundate'] = pd.to_datetime(bahan_db_df['Rundate'], dayfirst=True)\n",
    "\n",
    "# Convert the datetime objects back to strings in the desired format\n",
    "bahan_db_df['Rundate'] = bahan_db_df['Rundate'].dt.strftime('%Y/%m/%d')\n",
    "\n",
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Nucleotide [AAchange]</th>\n",
       "      <th>Clinical Relevance</th>\n",
       "      <th>Disease Name</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>P/LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "      <th>Variant_Record</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>4033732064895249566743264043910700766358748819...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>7632571583479047278483398426095059141725570125...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>6907276342077659380266749093569215599901803182...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>4902072944153808316336709269360209722627911761...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "      <td>3699264021093677716908640236982176456640991531...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>1117162511062382013085127344423789318027129222...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>7677422902686165563271874220330511463706534026...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>2095887827705408984261745191950751689122712668...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>1560944849085526669941227313946901905251088020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>3485652128672047435273636722318682603382369613...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Rundate     SampleID  \\\n",
       "0       2023/04/14  R2303061875   \n",
       "1       2023/04/14  R2303061875   \n",
       "2       2023/04/14  R2303061875   \n",
       "3       2023/04/14  R2303061875   \n",
       "4       2023/04/14  R2303061875   \n",
       "...            ...          ...   \n",
       "162647  2024/01/17  R2401062088   \n",
       "162648  2024/01/17  R2401062088   \n",
       "162649  2024/01/17  R2401062088   \n",
       "162650  2024/01/17  R2401062088   \n",
       "162651  2024/01/17  R2401062088   \n",
       "\n",
       "                                    Nucleotide [AAchange] Clinical Relevance  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--                      \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--                      \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--                      \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A             Benign   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--             Benign   \n",
       "...                                                   ...                ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--      Likely_benign   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--                      \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--                      \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--                      \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...                      \n",
       "\n",
       "                         Disease Name Chromosome    dbSNP ID   Gene     Start  \\\n",
       "0       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17032891   \n",
       "1       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044657   \n",
       "2       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044999   \n",
       "3       Paragangliomas 4 (MIM 115310)       chr1   rs2746462   SDHB  17054002   \n",
       "4          Pilomatrixoma (MIM 132600)       chr1   rs3219493  MUTYH  45330597   \n",
       "...                               ...        ...         ...    ...       ...   \n",
       "162647                                     chr22  rs17886236  CHEK2  28694987   \n",
       "162648                                     chr22              CHEK2  28695049   \n",
       "162649                                     chr22              CHEK2  28703766   \n",
       "162650                                     chr22              CHEK2  28734312   \n",
       "162651                                     chr22              CHEK2  28734825   \n",
       "\n",
       "             End Ref Alt Variant Type Variant_Classification     VAF Genotype  \\\n",
       "0       17032891   T   C          SNP                 Intron   49.74      Het   \n",
       "1       17044657   T   C          SNP                 Intron   45.91      Het   \n",
       "2       17044999   C   T          SNP                 Intron   44.70      Het   \n",
       "3       17054002   G   T          SNP                 Silent   99.66   HomAlt   \n",
       "4       45330597   G   C          SNP                 Intron  100.00   HomAlt   \n",
       "...          ...  ..  ..          ...                    ...     ...      ...   \n",
       "162647  28694987   C   G          SNP                 Intron   68.18      Het   \n",
       "162648  28695049   G   C          SNP                 Intron   58.62      Het   \n",
       "162649  28703766   T   C          SNP                 Intron   46.67      Het   \n",
       "162650  28734312   C   T          SNP                 Intron   31.71      Het   \n",
       "162651  28734826   -   A          INS                 Intron   50.00      Het   \n",
       "\n",
       "        alt_count  ref_count   DP   P/LP_Result Mode_of_Inheritance  \\\n",
       "0              96         97  193  not detected                  AD   \n",
       "1              73         86  159  not detected                  AD   \n",
       "2              97        120  217  not detected                  AD   \n",
       "3             876          3  879  not detected                  AD   \n",
       "4             545          0  545  not detected                  AR   \n",
       "...           ...        ...  ...           ...                 ...   \n",
       "162647         30         14   44  not detected                       \n",
       "162648         51         36   87  not detected                       \n",
       "162649          7          8   15  not detected                       \n",
       "162650         13         28   41  not detected                       \n",
       "162651         13         13   26  not detected                       \n",
       "\n",
       "                                           Variant_Record  \n",
       "0       4033732064895249566743264043910700766358748819...  \n",
       "1       7632571583479047278483398426095059141725570125...  \n",
       "2       6907276342077659380266749093569215599901803182...  \n",
       "3       4902072944153808316336709269360209722627911761...  \n",
       "4       3699264021093677716908640236982176456640991531...  \n",
       "...                                                   ...  \n",
       "162647  1117162511062382013085127344423789318027129222...  \n",
       "162648  7677422902686165563271874220330511463706534026...  \n",
       "162649  2095887827705408984261745191950751689122712668...  \n",
       "162650  1560944849085526669941227313946901905251088020...  \n",
       "162651  3485652128672047435273636722318682603382369613...  \n",
       "\n",
       "[162652 rows x 22 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bahan_db_df['Variant_Record'] = bahan_db_df.apply(lambda row: generate_unique_id(row['Rundate'] + ' | ' + row['SampleID'] + ' | ' + row['Nucleotide [AAchange]']), axis=1)\n",
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Nucleotide [AAchange]</th>\n",
       "      <th>Clinical Relevance</th>\n",
       "      <th>Disease Name</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>dbSNP ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>P/LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "      <th>Variant_Record</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>4033732064895249566743264043910700766358748819...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>7632571583479047278483398426095059141725570125...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>6907276342077659380266749093569215599901803182...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>4902072944153808316336709269360209722627911761...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "      <td>3699264021093677716908640236982176456640991531...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>1117162511062382013085127344423789318027129222...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>7677422902686165563271874220330511463706534026...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>2095887827705408984261745191950751689122712668...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>1560944849085526669941227313946901905251088020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>3485652128672047435273636722318682603382369613...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Rundate     SampleID  \\\n",
       "0       2023/04/14  R2303061875   \n",
       "1       2023/04/14  R2303061875   \n",
       "2       2023/04/14  R2303061875   \n",
       "3       2023/04/14  R2303061875   \n",
       "4       2023/04/14  R2303061875   \n",
       "...            ...          ...   \n",
       "162647  2024/01/17  R2401062088   \n",
       "162648  2024/01/17  R2401062088   \n",
       "162649  2024/01/17  R2401062088   \n",
       "162650  2024/01/17  R2401062088   \n",
       "162651  2024/01/17  R2401062088   \n",
       "\n",
       "                                    Nucleotide [AAchange] Clinical Relevance  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--                      \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--                      \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--                      \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A             Benign   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--             Benign   \n",
       "...                                                   ...                ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--      Likely_benign   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--                      \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--                      \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--                      \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...                      \n",
       "\n",
       "                         Disease Name Chromosome    dbSNP ID   Gene     Start  \\\n",
       "0       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17032891   \n",
       "1       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044657   \n",
       "2       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044999   \n",
       "3       Paragangliomas 4 (MIM 115310)       chr1   rs2746462   SDHB  17054002   \n",
       "4          Pilomatrixoma (MIM 132600)       chr1   rs3219493  MUTYH  45330597   \n",
       "...                               ...        ...         ...    ...       ...   \n",
       "162647                                     chr22  rs17886236  CHEK2  28694987   \n",
       "162648                                     chr22              CHEK2  28695049   \n",
       "162649                                     chr22              CHEK2  28703766   \n",
       "162650                                     chr22              CHEK2  28734312   \n",
       "162651                                     chr22              CHEK2  28734825   \n",
       "\n",
       "             End Ref Alt Variant Type Variant_Classification     VAF Genotype  \\\n",
       "0       17032891   T   C          SNP                 Intron   49.74      Het   \n",
       "1       17044657   T   C          SNP                 Intron   45.91      Het   \n",
       "2       17044999   C   T          SNP                 Intron   44.70      Het   \n",
       "3       17054002   G   T          SNP                 Silent   99.66   HomAlt   \n",
       "4       45330597   G   C          SNP                 Intron  100.00   HomAlt   \n",
       "...          ...  ..  ..          ...                    ...     ...      ...   \n",
       "162647  28694987   C   G          SNP                 Intron   68.18      Het   \n",
       "162648  28695049   G   C          SNP                 Intron   58.62      Het   \n",
       "162649  28703766   T   C          SNP                 Intron   46.67      Het   \n",
       "162650  28734312   C   T          SNP                 Intron   31.71      Het   \n",
       "162651  28734826   -   A          INS                 Intron   50.00      Het   \n",
       "\n",
       "        alt_count  ref_count   DP   P/LP_Result Mode_of_Inheritance  \\\n",
       "0              96         97  193  not detected                  AD   \n",
       "1              73         86  159  not detected                  AD   \n",
       "2              97        120  217  not detected                  AD   \n",
       "3             876          3  879  not detected                  AD   \n",
       "4             545          0  545  not detected                  AR   \n",
       "...           ...        ...  ...           ...                 ...   \n",
       "162647         30         14   44  not detected                       \n",
       "162648         51         36   87  not detected                       \n",
       "162649          7          8   15  not detected                       \n",
       "162650         13         28   41  not detected                       \n",
       "162651         13         13   26  not detected                       \n",
       "\n",
       "                                           Variant_Record  \n",
       "0       4033732064895249566743264043910700766358748819...  \n",
       "1       7632571583479047278483398426095059141725570125...  \n",
       "2       6907276342077659380266749093569215599901803182...  \n",
       "3       4902072944153808316336709269360209722627911761...  \n",
       "4       3699264021093677716908640236982176456640991531...  \n",
       "...                                                   ...  \n",
       "162647  1117162511062382013085127344423789318027129222...  \n",
       "162648  7677422902686165563271874220330511463706534026...  \n",
       "162649  2095887827705408984261745191950751689122712668...  \n",
       "162650  1560944849085526669941227313946901905251088020...  \n",
       "162651  3485652128672047435273636722318682603382369613...  \n",
       "\n",
       "[162652 rows x 22 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           Rundate     SampleID  \\\n",
      "29274   2023/03/06  R2303041918   \n",
      "37462   2024/01/19  R2401131933   \n",
      "43808   2023/10/16  R2310100825   \n",
      "44194   2023/10/16  R2310121163   \n",
      "152620  2021/07/07         HCP1   \n",
      "152898  2021/07/07         HCP1   \n",
      "\n",
      "                                   Nucleotide [AAchange] Clinical Relevance  \\\n",
      "29274   __UNKNOWN__|NM_001243998|__UNKNOWN__|__UNKNOWN__                      \n",
      "37462      __UNKNOWN__|NM_000249|__UNKNOWN__|__UNKNOWN__                      \n",
      "43808   __UNKNOWN__|NM_001126049|__UNKNOWN__|__UNKNOWN__             Benign   \n",
      "44194      __UNKNOWN__|NM_000249|__UNKNOWN__|__UNKNOWN__                      \n",
      "152620                         __UNKNOWN__ [__UNKNOWN__]                      \n",
      "152898                         __UNKNOWN__ [__UNKNOWN__]                      \n",
      "\n",
      "                        Disease Name   Chromosome   dbSNP ID         Gene  \\\n",
      "29274                                 __UNKNOWN__             __UNKNOWN__   \n",
      "37462   Lynch syndrome (MIM 120435)   __UNKNOWN__             __UNKNOWN__   \n",
      "43808                                 __UNKNOWN__  rs2943772  __UNKNOWN__   \n",
      "44194   Lynch syndrome (MIM 120435)   __UNKNOWN__             __UNKNOWN__   \n",
      "152620  Lynch syndrome (MIM 120435)   __UNKNOWN__             __UNKNOWN__   \n",
      "152898                                __UNKNOWN__             __UNKNOWN__   \n",
      "\n",
      "              Start          End          Ref          Alt Variant Type  \\\n",
      "29274   __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__   \n",
      "37462   __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__   \n",
      "43808   __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__   \n",
      "44194   __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__  __UNKNOWN__   \n",
      "152620  __UNKNOWN__  __UNKNOWN__                            __UNKNOWN__   \n",
      "152898  __UNKNOWN__  __UNKNOWN__                            __UNKNOWN__   \n",
      "\n",
      "       Variant_Classification     VAF Genotype  alt_count  ref_count   DP  \\\n",
      "29274             __UNKNOWN__  100.00   HomAlt          2          0    2   \n",
      "37462             __UNKNOWN__  100.00   HomAlt         10          0   10   \n",
      "43808             __UNKNOWN__  100.00   HomAlt         53          0   53   \n",
      "44194             __UNKNOWN__  100.00   HomAlt         14          0   14   \n",
      "152620            __UNKNOWN__   94.12   HomAlt         16          1  196   \n",
      "152898            __UNKNOWN__  100.00   HomAlt          3          0   12   \n",
      "\n",
      "         P/LP_Result Mode_of_Inheritance  \\\n",
      "29274   not detected                       \n",
      "37462   not detected                  AD   \n",
      "43808   not detected                       \n",
      "44194   not detected                  AD   \n",
      "152620      detected                  AD   \n",
      "152898      detected                       \n",
      "\n",
      "                                           Variant_Record  \n",
      "29274   8154310245376185821562886404894591733365170785...  \n",
      "37462   9755085109107587011234098115848160271409611505...  \n",
      "43808   3976776801748119458805457232535362159087700947...  \n",
      "44194   2429434306197402874676997484355056785253788137...  \n",
      "152620  8136120995176728916552532255382322083062747056...  \n",
      "152898  8136120995176728916552532255382322083062747056...  \n"
     ]
    }
   ],
   "source": [
    "bahan_db_df.at[0,'Variant_Record']\n",
    "duplicate_rows_specific = bahan_db_df[bahan_db_df.duplicated(['Variant_Record'])]\n",
    "print(duplicate_rows_specific)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rundate</th>\n",
       "      <th>SampleID</th>\n",
       "      <th>Variant</th>\n",
       "      <th>Clinical_Relevance</th>\n",
       "      <th>Disease_Name</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>dbSNP_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>alt_count</th>\n",
       "      <th>ref_count</th>\n",
       "      <th>DP</th>\n",
       "      <th>P_LP_Result</th>\n",
       "      <th>Mode_of_Inheritance</th>\n",
       "      <th>Variant_Record</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17032891T&gt;C|NM_003000|c.e3-169A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17032891</td>\n",
       "      <td>17032891</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>49.74</td>\n",
       "      <td>Het</td>\n",
       "      <td>96</td>\n",
       "      <td>97</td>\n",
       "      <td>193</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>4033732064895249566743264043910700766358748819...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044657T&gt;C|NM_003000|c.e2-104A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044657</td>\n",
       "      <td>17044657</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>45.91</td>\n",
       "      <td>Het</td>\n",
       "      <td>73</td>\n",
       "      <td>86</td>\n",
       "      <td>159</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>7632571583479047278483398426095059141725570125...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17044999C&gt;T|NM_003000|c.e2+111G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td></td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17044999</td>\n",
       "      <td>17044999</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>44.70</td>\n",
       "      <td>Het</td>\n",
       "      <td>97</td>\n",
       "      <td>120</td>\n",
       "      <td>217</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>6907276342077659380266749093569215599901803182...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:17054002G&gt;T|NM_003000|c.18C&gt;A|p.A6A</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Paragangliomas 4 (MIM 115310)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs2746462</td>\n",
       "      <td>SDHB</td>\n",
       "      <td>17054002</td>\n",
       "      <td>17054002</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Silent</td>\n",
       "      <td>99.66</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>876</td>\n",
       "      <td>3</td>\n",
       "      <td>879</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AD</td>\n",
       "      <td>4902072944153808316336709269360209722627911761...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2023/04/14</td>\n",
       "      <td>R2303061875</td>\n",
       "      <td>g.chr1:45330597G&gt;C|NM_001048171|c.e3+1160C&gt;G|--</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Pilomatrixoma (MIM 132600)</td>\n",
       "      <td>chr1</td>\n",
       "      <td>rs3219493</td>\n",
       "      <td>MUTYH</td>\n",
       "      <td>45330597</td>\n",
       "      <td>45330597</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>100.00</td>\n",
       "      <td>HomAlt</td>\n",
       "      <td>545</td>\n",
       "      <td>0</td>\n",
       "      <td>545</td>\n",
       "      <td>not detected</td>\n",
       "      <td>AR</td>\n",
       "      <td>3699264021093677716908640236982176456640991531...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162647</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28694987C&gt;G|NM_001005735|c.e13-140G&gt;C|--</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td>rs17886236</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28694987</td>\n",
       "      <td>28694987</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>68.18</td>\n",
       "      <td>Het</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>44</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>1117162511062382013085127344423789318027129222...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162648</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28695049G&gt;C|NM_001005735|c.e13-78C&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28695049</td>\n",
       "      <td>28695049</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>58.62</td>\n",
       "      <td>Het</td>\n",
       "      <td>51</td>\n",
       "      <td>36</td>\n",
       "      <td>87</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>7677422902686165563271874220330511463706534026...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162649</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28703766T&gt;C|NM_001005735|c.e9+200A&gt;G|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28703766</td>\n",
       "      <td>28703766</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>46.67</td>\n",
       "      <td>Het</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>2095887827705408984261745191950751689122712668...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162650</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734312C&gt;T|NM_001005735|c.e3-91G&gt;A|--</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734312</td>\n",
       "      <td>28734312</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Intron</td>\n",
       "      <td>31.71</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>28</td>\n",
       "      <td>41</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>1560944849085526669941227313946901905251088020...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162651</th>\n",
       "      <td>2024/01/17</td>\n",
       "      <td>R2401062088</td>\n",
       "      <td>g.chr22:28734825_28734826insA|NM_001005735|c.e...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>chr22</td>\n",
       "      <td></td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>28734825</td>\n",
       "      <td>28734826</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>INS</td>\n",
       "      <td>Intron</td>\n",
       "      <td>50.00</td>\n",
       "      <td>Het</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>not detected</td>\n",
       "      <td></td>\n",
       "      <td>3485652128672047435273636722318682603382369613...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>162652 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Rundate     SampleID  \\\n",
       "0       2023/04/14  R2303061875   \n",
       "1       2023/04/14  R2303061875   \n",
       "2       2023/04/14  R2303061875   \n",
       "3       2023/04/14  R2303061875   \n",
       "4       2023/04/14  R2303061875   \n",
       "...            ...          ...   \n",
       "162647  2024/01/17  R2401062088   \n",
       "162648  2024/01/17  R2401062088   \n",
       "162649  2024/01/17  R2401062088   \n",
       "162650  2024/01/17  R2401062088   \n",
       "162651  2024/01/17  R2401062088   \n",
       "\n",
       "                                                  Variant Clinical_Relevance  \\\n",
       "0             g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--                      \n",
       "1             g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--                      \n",
       "2             g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--                      \n",
       "3              g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A             Benign   \n",
       "4         g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--             Benign   \n",
       "...                                                   ...                ...   \n",
       "162647   g.chr22:28694987C>G|NM_001005735|c.e13-140G>C|--      Likely_benign   \n",
       "162648    g.chr22:28695049G>C|NM_001005735|c.e13-78C>G|--                      \n",
       "162649    g.chr22:28703766T>C|NM_001005735|c.e9+200A>G|--                      \n",
       "162650     g.chr22:28734312C>T|NM_001005735|c.e3-91G>A|--                      \n",
       "162651  g.chr22:28734825_28734826insA|NM_001005735|c.e...                      \n",
       "\n",
       "                         Disease_Name Chromosome    dbSNP_ID   Gene     Start  \\\n",
       "0       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17032891   \n",
       "1       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044657   \n",
       "2       Paragangliomas 4 (MIM 115310)       chr1               SDHB  17044999   \n",
       "3       Paragangliomas 4 (MIM 115310)       chr1   rs2746462   SDHB  17054002   \n",
       "4          Pilomatrixoma (MIM 132600)       chr1   rs3219493  MUTYH  45330597   \n",
       "...                               ...        ...         ...    ...       ...   \n",
       "162647                                     chr22  rs17886236  CHEK2  28694987   \n",
       "162648                                     chr22              CHEK2  28695049   \n",
       "162649                                     chr22              CHEK2  28703766   \n",
       "162650                                     chr22              CHEK2  28734312   \n",
       "162651                                     chr22              CHEK2  28734825   \n",
       "\n",
       "             End Ref Alt Variant_Type Variant_Classification     VAF Genotype  \\\n",
       "0       17032891   T   C          SNP                 Intron   49.74      Het   \n",
       "1       17044657   T   C          SNP                 Intron   45.91      Het   \n",
       "2       17044999   C   T          SNP                 Intron   44.70      Het   \n",
       "3       17054002   G   T          SNP                 Silent   99.66   HomAlt   \n",
       "4       45330597   G   C          SNP                 Intron  100.00   HomAlt   \n",
       "...          ...  ..  ..          ...                    ...     ...      ...   \n",
       "162647  28694987   C   G          SNP                 Intron   68.18      Het   \n",
       "162648  28695049   G   C          SNP                 Intron   58.62      Het   \n",
       "162649  28703766   T   C          SNP                 Intron   46.67      Het   \n",
       "162650  28734312   C   T          SNP                 Intron   31.71      Het   \n",
       "162651  28734826   -   A          INS                 Intron   50.00      Het   \n",
       "\n",
       "        alt_count  ref_count   DP   P_LP_Result Mode_of_Inheritance  \\\n",
       "0              96         97  193  not detected                  AD   \n",
       "1              73         86  159  not detected                  AD   \n",
       "2              97        120  217  not detected                  AD   \n",
       "3             876          3  879  not detected                  AD   \n",
       "4             545          0  545  not detected                  AR   \n",
       "...           ...        ...  ...           ...                 ...   \n",
       "162647         30         14   44  not detected                       \n",
       "162648         51         36   87  not detected                       \n",
       "162649          7          8   15  not detected                       \n",
       "162650         13         28   41  not detected                       \n",
       "162651         13         13   26  not detected                       \n",
       "\n",
       "                                           Variant_Record  \n",
       "0       4033732064895249566743264043910700766358748819...  \n",
       "1       7632571583479047278483398426095059141725570125...  \n",
       "2       6907276342077659380266749093569215599901803182...  \n",
       "3       4902072944153808316336709269360209722627911761...  \n",
       "4       3699264021093677716908640236982176456640991531...  \n",
       "...                                                   ...  \n",
       "162647  1117162511062382013085127344423789318027129222...  \n",
       "162648  7677422902686165563271874220330511463706534026...  \n",
       "162649  2095887827705408984261745191950751689122712668...  \n",
       "162650  1560944849085526669941227313946901905251088020...  \n",
       "162651  3485652128672047435273636722318682603382369613...  \n",
       "\n",
       "[162652 rows x 22 columns]"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_column_name = {\n",
    "    'Nucleotide [AAchange]':'Variant',\n",
    "    'Clinical Relevance':'Clinical_Relevance',\n",
    "    'Disease Name':'Disease_Name',\n",
    "    'dbSNP ID':'dbSNP_ID',\n",
    "    'Variant Type':'Variant_Type',\n",
    "    'P/LP_Result':'P_LP_Result'\n",
    "}\n",
    "bahan_db_df.rename(columns=final_column_name, inplace=True)\n",
    "bahan_db_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "# SQL statement for creating a table\n",
    "create_table_sql1 = f\"\"\"\n",
    "CREATE TABLE IF NOT EXISTS {table_name1} (\n",
    "    `Rundate` TEXT,\n",
    "    `SampleID` TEXT,\n",
    "    `Variant` TEXT,\n",
    "    `Clinical_Relevance` TEXT,\n",
    "    `Disease_Name` TEXT,\n",
    "    `Chromosome` TEXT,\n",
    "    `dbSNP_ID` TEXT,\n",
    "    `Gene` TEXT,\n",
    "    `Start` TEXT,\n",
    "    `End` TEXT,\n",
    "    `Ref` TEXT,\n",
    "    `Alt` TEXT,\n",
    "    `Variant_Type` TEXT,\n",
    "    `Variant_Classification` TEXT,\n",
    "    `VAF` TEXT,\n",
    "    `Genotype` TEXT,\n",
    "    `alt_count` TEXT,\n",
    "    `ref_count` TEXT,\n",
    "    `DP` TEXT,\n",
    "    `P_LP_Result` TEXT,\n",
    "    `Mode_of_Inheritance` TEXT,\n",
    "    `Variant_Record` TEXT PRIMARY KEY\n",
    ");\n",
    "\"\"\"\n",
    "\n",
    "# SQL statement for creating a table\n",
    "create_table_sql2 = f\"\"\"\n",
    "CREATE TABLE IF NOT EXISTS {table_name2} (\n",
    "    `version_id` INTEGER PRIMARY KEY AUTOINCREMENT,\n",
    "    `Rundate` TEXT,\n",
    "    `SampleID` TEXT,\n",
    "    `Variant` TEXT,\n",
    "    `Clinical_Relevance` TEXT,\n",
    "    `Disease_Name` TEXT,\n",
    "    `Chromosome` TEXT,\n",
    "    `dbSNP_ID` TEXT,\n",
    "    `Gene` TEXT,\n",
    "    `Start` TEXT,\n",
    "    `End` TEXT,\n",
    "    `Ref` TEXT,\n",
    "    `Alt` TEXT,\n",
    "    `Variant_Type` TEXT,\n",
    "    `Variant_Classification` TEXT,\n",
    "    `VAF` TEXT,\n",
    "    `Genotype` TEXT,\n",
    "    `alt_count` TEXT,\n",
    "    `ref_count` TEXT,\n",
    "    `DP` TEXT,\n",
    "    `P_LP_Result` TEXT,\n",
    "    `Mode_of_Inheritance` TEXT,\n",
    "    `Variant_Record` TEXT,\n",
    "    `changed_by` TEXT,\n",
    "    `change_date` TIMESTAMP DEFAULT CURRENT_TIMESTAMP,\n",
    "    FOREIGN KEY(`Variant_Record`) REFERENCES {table_name1} (`Variant_Record`)\n",
    ");\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('2023/04/14', 'R2303061875', 'g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--', ' ', 'Paragangliomas 4 (MIM 115310)', 'chr1', ' ', 'SDHB', '17032891', '17032891', 'T', 'C', 'SNP', 'Intron', '49.74', 'Het', '96', '97', '193', 'not detected', 'AD', '40337320648952495667432640439107007663587488191788217981741751870061962988587')\n",
      "('2023/04/14', 'R2303061875', 'g.chr1:17044657T>C|NM_003000|c.e2-104A>G|--', ' ', 'Paragangliomas 4 (MIM 115310)', 'chr1', ' ', 'SDHB', '17044657', '17044657', 'T', 'C', 'SNP', 'Intron', '45.91', 'Het', '73', '86', '159', 'not detected', 'AD', '7632571583479047278483398426095059141725570125641311441897613403591148550905')\n",
      "('2023/04/14', 'R2303061875', 'g.chr1:17044999C>T|NM_003000|c.e2+111G>A|--', ' ', 'Paragangliomas 4 (MIM 115310)', 'chr1', ' ', 'SDHB', '17044999', '17044999', 'C', 'T', 'SNP', 'Intron', '44.7', 'Het', '97', '120', '217', 'not detected', 'AD', '69072763420776593802667490935692155999018031822586539961196970907651305157646')\n",
      "('2023/04/14', 'R2303061875', 'g.chr1:17054002G>T|NM_003000|c.18C>A|p.A6A', 'Benign', 'Paragangliomas 4 (MIM 115310)', 'chr1', 'rs2746462', 'SDHB', '17054002', '17054002', 'G', 'T', 'SNP', 'Silent', '99.66', 'HomAlt', '876', '3', '879', 'not detected', 'AD', '49020729441538083163367092693602097226279117613836441905661411210270365229035')\n",
      "('2023/04/14', 'R2303061875', 'g.chr1:45330597G>C|NM_001048171|c.e3+1160C>G|--', 'Benign', 'Pilomatrixoma (MIM 132600)', 'chr1', 'rs3219493', 'MUTYH', '45330597', '45330597', 'G', 'C', 'SNP', 'Intron', '100.0', 'HomAlt', '545', '0', '545', 'not detected', 'AR', '36992640210936777169086402369821764566409915316346923331999745901548428032102')\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import sqlite3\n",
    "\n",
    "\n",
    "vava_db_dir = '/home/sysadmin/vava_db/'\n",
    "database_file = 'TSHCvar.db'\n",
    "\n",
    "\n",
    "# Remove duplicates based on the 'Variant Record' column, keeping the first occurrence\n",
    "df_unique = bahan_db_df.drop_duplicates(subset=['Variant_Record'], keep='first')\n",
    "\n",
    "\n",
    "table_name1 = 'db_var'\n",
    "table_name2 = 'db_var_version_history'\n",
    "\n",
    "# Replace 'your_database.sqlite' with the path to your SQLite file\n",
    "db_path = f'{vava_db_dir}/{database_file}'\n",
    "\n",
    "# Connect to the SQLite database\n",
    "conn = sqlite3.connect(db_path)\n",
    "cursor = conn.cursor()\n",
    "\n",
    "\n",
    "\n",
    "# Execute the SQL statement to create the table\n",
    "cursor.execute(create_table_sql1)\n",
    "cursor.execute(create_table_sql2)\n",
    "\n",
    "# Insert the deduplicated data into the SQL table\n",
    "df_unique.to_sql(table_name1, conn, if_exists='append', index=False, method='multi', chunksize=500)\n",
    "\n",
    "# Verify insertion by querying the first 5 rows\n",
    "cursor.execute(f\"SELECT * FROM {table_name1} LIMIT 5;\")\n",
    "rows = cursor.fetchall()\n",
    "\n",
    "# Display the first 5 rows from the database for verification\n",
    "for row in rows:\n",
    "    print(row)\n",
    "\n",
    "# Verify insertion by querying the first 5 rows\n",
    "cursor.execute(f\"SELECT * FROM {table_name2} LIMIT 5;\")\n",
    "rows = cursor.fetchall()\n",
    "\n",
    "# Display the first 5 rows from the database for verification\n",
    "for row in rows:\n",
    "    print(row)\n",
    "\n",
    "\n",
    "# Close the connection to the database\n",
    "conn.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('2023/04/14', 'R2303061875', 'g.chr1:17032891T>C|NM_003000|c.e3-169A>G|--', ' ', 'Paragangliomas 4 (MIM 115310)', 'chr1', ' ', 'SDHB', '17032891', '17032891', 'T', 'C', 'SNP', 'Intron', '49.74', 'Het', '96', '97', '193', 'not detected', 'AD', '40337320648952495667432640439107007663587488191788217981741751870061962988587')\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import sqlite3\n",
    "\n",
    "\n",
    "vava_db_dir = '/home/sysadmin/vava_db/'\n",
    "database_file = 'TSHCvar_empty.db'\n",
    "\n",
    "\n",
    "# Remove duplicates based on the 'Variant Record' column, keeping the first occurrence\n",
    "df_unique = bahan_db_df.drop_duplicates(subset=['Variant_Record'], keep='first')\n",
    "dummy_df = df_unique.head(1)\n",
    "dummy_df\n",
    "\n",
    "table_name1 = 'db_var'\n",
    "table_name2 = 'db_var_version_history'\n",
    "\n",
    "# Replace 'your_database.sqlite' with the path to your SQLite file\n",
    "db_path = f'{vava_db_dir}/{database_file}'\n",
    "\n",
    "# Connect to the SQLite database\n",
    "conn = sqlite3.connect(db_path)\n",
    "cursor = conn.cursor()\n",
    "\n",
    "\n",
    "# Execute the SQL statement to create the table\n",
    "cursor.execute(create_table_sql1)\n",
    "cursor.execute(create_table_sql2)\n",
    "\n",
    "# Insert the deduplicated data into the SQL table\n",
    "dummy_df.to_sql(table_name1, conn, if_exists='append', index=False, method='multi', chunksize=500)\n",
    "\n",
    "# Verify insertion by querying the first 5 rows\n",
    "cursor.execute(f\"SELECT * FROM {table_name1} LIMIT 5;\")\n",
    "rows = cursor.fetchall()\n",
    "\n",
    "# Display the first 5 rows from the database for verification\n",
    "for row in rows:\n",
    "    print(row)\n",
    "\n",
    "# Verify insertion by querying the first 5 rows\n",
    "cursor.execute(f\"SELECT * FROM {table_name2} LIMIT 5;\")\n",
    "rows = cursor.fetchall()\n",
    "\n",
    "# Display the first 5 rows from the database for verification\n",
    "for row in rows:\n",
    "    print(row)\n",
    "\n",
    "\n",
    "# Close the connection to the database\n",
    "conn.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_to_excel('/mnt/mol/test_db.xlsx', dfs={'Sheet1':bahan_db_df})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "`Rundate`, `SampleID`, `Nucleotide [AAchange]`, `Clinical Relevance`, `Disease Name`, `Chromosome`, `dbSNP ID`, `Gene`, `Start`, `End`, `Ref`, `Alt`, `Variant Type`, `Variant_Classification`, `VAF`, `Genotype`, `alt_count`, `ref_count`, `DP`, `P/LP_Result`, `Mode_of_Inheritance`, `Variant_Record`, `changed_by`"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
